packages/widget/src/pages/SwapPage/SwapPage.tsx

```diff
@@ -50,6 +57,19 @@ export const SwapPage: React.FC = () => {
 
   const handleExecuteRoute = useCallback(() => {
     if (tokenValueBottomSheetRef.current?.isOpen()) {
+      if (route) {
+        emitter.emit(WidgetEvent.RouteHighValueLoss, {
+          fromAmountUsd: route.fromAmountUSD,
+          gasCostUSD: route.gasCostUSD,
+          toAmountUSD: route.toAmountUSD,
+          valueLoss: Big(route.toAmountUSD || 0)
+            .div(Big(route.fromAmountUSD || 0).plus(Big(route.gasCostUSD || 0)))
+            .minus(1)
+            .mul(100)
+            .round(2, Big.roundUp)
+            .toString(),
+        });
```

Comment:
How about combining the creation of this object to some function and using both inside TokenValueLossBottomSheet component and here? 🙂

Reply:
Good idea, and DRYer!

Reply:
@chybisov I have moved that piece of code to re-use. It´s to be found in the SwapPage.tsx named **calcValueLoss**. Not sure if a better place would be the utils-folder?

Reply:
> @chybisov I have moved that piece of code to re-use. It´s to be found in the SwapPage.tsx named **calcValueLoss**. Not sure if a better place would be the utils-folder?

Yes, please, create a utils.ts inside the SwapPage folder.